File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        Map<String, Integer> marathon = new HashMap<>();
        for(String s : participant) marathon.put(s, marathon.getOrDefault(s,0)+1);
        for(String s : completion) marathon.put(s, marathon.get(s)-1);
        for(Map.Entry<String, Integer> e : marathon.entrySet()) {
            if(e.getValue()==1) {
                answer = e.getKey();
                break;
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <bits/stdc++.h>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    unordered_map<string,int> m;
    for(string p : participant) m[p]++;
    for(string c : completion) m[c]--;
    for(auto it : m) {
        if(it.second>0) return it.first;
    }
}